### PR TITLE
fixed issue of new user able to see previous user messages

### LINKED ISSUE #16: 2nd ISSUE
🎉 **The Most Awaited Second Issue is Here!** 🎉

🔥 Introducing the **`server.py` Update Challenge** for our chat app, Mirage! We're looking for contributors to help evolve the server. Please keep these points in mind before diving in:

### 🚀 **What We're Looking For:**
1. **Flask Framework:** The current `server.py` uses Flask. If you decide to proceed with Flask, there’s no need to modify `client.py`.
2. **SocketIO Transition:** Many contributors have shown interest in switching to SocketIO. If you choose this path, we'd appreciate it if you also update `client.py` accordingly to maintain compatibility.

### 💡 **Key Features to Retain:**
These are the core aspects that make Mirage unique. You're free to **improvise** and make things better—just keep the essence intact!
1. Messages are **temporarily stored** on the server until all clients receive them, and then they're **deleted**.
2. All data (like rooms and user counts) is stored in a **JSON file**—no databases, keeping it simple and lightweight.
3. No **authentication** is involved now, nor is it planned for the future.
4. **No chat logs**—privacy is a core feature.

Feel free to enhance anything else while maintaining the current functionality!

---

**🔍 PLEASE CHECK YOUR CODE BEFORE SUBMITTING A PR**

Thank you so much for your contributions. Enjoy Hacktoberfest! 🌿✨

> 🚨 Note: The **client UI modification** issue is still open. If a server PR is accepted, the UI issue will be closed for now and reopened later.

🔒 **Security features integration** will be the next exciting challenge once the server updates are complete, so stay tuned!

🌟 **Don't forget to star the repo!** 🌟

### PR DESCRIPTION
**Pull request by eleensmathew**
fixed issue of new user able to see previous user messages. Fixes #16

### CODE REVIEW

File: server.py [hunk 1]
```diff
@@ -45,9 +49,12 @@ def leave_room(username, room_name):
 
 # Route to get all rooms
 @app.route('/get_rooms', methods=['GET'])
-def get_rooms():
+def get_rooms(): #only returns room of the specefic user
     rooms = load_rooms()
-    return jsonify(list(rooms.keys()))
+    username = get_username()
+    user_rooms = {room_id: users for room_id, users in rooms.items() if username in users}
+    p
```
Comment (eleensmathew) [inline server.py:56]: will remove the p

Comment (KorryKatti) [reply server.py]: can you elaborate on what you have changed

**Comment by eleensmathew** [comment]
in get_rooms I have only fetched the rooms which the user is part of and not all.
And I also added a timestamp to the message field so that older messages can be deleted.
Delete_old_message function deletes the older messages of that particular room

**Comment by eleensmathew** [comment]
Also if your running the server in your device you would have to delete all the previous messages sunce those messages dont have a timestamp.

**Comment by eleensmathew** [comment]
I can also add the functionality of encryption and web socket as well to this pr.

**Comment by KorryKatti** [comment]
LGTM

**Pull request merged**